warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -262,6 +263,17 @@ public B withIvaratorSourcePool(GenericObjectPool<SortedKeyValueIterator<Key,Val
     // a time filter
     private final TimeFilter timeFilter;
     
+    // construct selective bounding ranges if handed a document range
+    protected String uid;
+    protected String datatype;
+    // support for determining if we are in a high cardinality field value
+    protected long consecutiveFieldValueHits = 0L;
+    protected long consecutiveHitsThreshold = 40L; // based on intra-index block cost of a seek vs next
```

Comment:
should this be configurable?

Reply:
alternatively creating a ticket to make it configurable in the future would be fine

Reply:
Captured in https://github.com/NationalSecurityAgency/datawave/issues/1205